### ISSUE TITLE
Trouble with reshape()

### ISSUE THREAD

**Issue by NiklasGustafsson** [FluxML/Zygote.jl#187]
The [attached](https://github.com/FluxML/Zygote.jl/files/3175188/repro.jl.txt) naive version of the Transformer neural network architecture fails to generate a gradient function. The problem seems to be on line 39, in the softmax function.

If I can narrow this down to a smaller sample, I will update the issue with the smaller repro.
